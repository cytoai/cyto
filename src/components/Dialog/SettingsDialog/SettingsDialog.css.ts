import {createStyles} from "@material-ui/styles";
import {Theme} from "@material-ui/core";

const styles = (theme: Theme) => createStyles({
  collapsed: {
    backgroundColor: '#f5f5f5'
  },
  root: {
    flexGrow: 1
  },
  flex: {
    flex: 1
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120
  },
  appbar: {
    boxShadow: 'none'
  },
  secondary: {
    padding: 0
  }
});

export default styles;
