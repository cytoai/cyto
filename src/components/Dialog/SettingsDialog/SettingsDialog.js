import * as React from 'react';
import styles from './SettingsDialog.css';
import {
  AppBar,
  Collapse,
  Dialog,
  Divider,
  Grid,
  IconButton,
  List,
  ListItem,
  ListItemSecondaryAction,
  ListItemText,
  Switch,
  Toolbar,
  Typography
} from '@material-ui/core';
import ExpandLessIcon from '@material-ui/icons/ExpandLess';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import { makeStyles } from '@material-ui/styles';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';

const useStyles = makeStyles(styles);

const SettingsDialog = props => {
  const [classificationCollapsed, setClassificationCollapsed] = React.useState(
    false
  );

  const [notificationsClicked, setNotificationsClicked] = React.useState(false);

  const { onClose, open } = props;

  const classes = useStyles();

  return (
    <Dialog className={classes.root} fullScreen open={open} onClose={onClose}>
      <div className={classes.root}>
        <AppBar position="static" color="inherit" className={classes.appbar}>
          <Toolbar>
            <IconButton
              className={classes.menuButton}
              color="inherit"
              aria-label="Menu"
              onClick={onClose}
            >
              <ArrowBackIcon />
            </IconButton>

            <Typography variant="h6" color="inherit" className={classes.flex}>
              Settings
            </Typography>
          </Toolbar>
        </AppBar>

        <Grid container spacing={24}>
          <Grid item xs={1} sm={1} md={3} lg={4} />

          <Grid item xs={10} sm={10} md={6} lg={4}>
            <List component="div">
              <ListItem>
                <Grid item xs={12}>
                  <ListItemText primary="Notifications" />
                  <ListItemText
                    className={classes.secondary}
                    secondary="Notifications are disabled. Learn more."
                  />
                </Grid>

                <ListItemSecondaryAction>
                  <Switch
                    onChange={() =>
                      setNotificationsClicked(!notificationsClicked)
                    }
                    checked={notificationsClicked}
                  />
                </ListItemSecondaryAction>
              </ListItem>
            </List>

            <Divider />

            <List component="nav">
              <ListItem
                button
                onClick={() =>
                  setClassificationCollapsed(!classificationCollapsed)
                }
              >
                <ListItemText primary="Object recognition" />

                {classificationCollapsed ? (
                  <ExpandLessIcon />
                ) : (
                  <ExpandMoreIcon />
                )}
              </ListItem>

              <Collapse
                in={!classificationCollapsed}
                timeout="auto"
                unmountOnExit
              >
                <List
                  component="div"
                  disablePadding
                  className={classes.collapsed}
                >
                  <ListItem>
                    <ListItemText primary="Loss function" />
                  </ListItem>

                  <ListItem>
                    <ListItemText primary="Learning rate" />
                  </ListItem>
                </List>
              </Collapse>
            </List>
          </Grid>
        </Grid>
      </div>
    </Dialog>
  );
};

export default SettingsDialog;
