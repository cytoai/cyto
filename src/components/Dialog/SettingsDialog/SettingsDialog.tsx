import * as React from 'react';
import styles from './SettingsDialog.css';
import {
  AppBar,
  Dialog,
  IconButton,
  Slide,
  Toolbar,
  Typography
} from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';

const useStyles = makeStyles(styles);

const SettingsDialogTransition = (props: any) => (
  <Slide direction="right" {...props} />
);

const SettingsDialog = (props: any) => {
  const { onClose, open } = props;

  const classes = useStyles();

  return (
    <Dialog
      className={classes.root}
      fullScreen
      open={open}
      onClose={onClose}
      TransitionComponent={SettingsDialogTransition}
    >
      <div className={classes.root}>
        <AppBar position="static" color="inherit" className={classes.appbar}>
          <Toolbar>
            <IconButton
              className={classes.menuButton}
              color="inherit"
              aria-label="Menu"
              onClick={onClose}
            >
              <ArrowBackIcon />
            </IconButton>

            <Typography variant="h6" color="inherit" className={classes.flex}>
              Settings
            </Typography>
          </Toolbar>
        </AppBar>
      </div>
    </Dialog>
  );
};

export default SettingsDialog;
