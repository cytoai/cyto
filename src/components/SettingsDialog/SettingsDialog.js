import React from 'react';
import styles from './SettingsDialog.css';
import { withStyles } from '@material-ui/core/styles';
import { Dialog } from '@material-ui/core';
import Settings from '../Settings/Settings';

function SettingsDialog(props) {
  const { classes, onClose, open } = props;

  return (
    <Dialog className={classes.root} fullScreen open={open} onClose={onClose}>
      <Settings onClose={onClose} />
    </Dialog>
  );
}

export default withStyles(styles, { withTheme: true })(SettingsDialog);
