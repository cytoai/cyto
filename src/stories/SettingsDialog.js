import React from 'react';
import { storiesOf } from '@storybook/react';
import SettingsDialog from '../components/Dialog/SettingsDialog/SettingsDialog';

storiesOf('SettingsDialog', module).add('default', () => <SettingsDialog />);
