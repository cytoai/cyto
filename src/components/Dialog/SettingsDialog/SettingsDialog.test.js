import React from 'react';
import { shallow } from 'enzyme';
import SettingsDialog from './SettingsDialog';

describe('SettingsDialog', () => {
  it('should render correctly', () => {
    const component = shallow(<SettingsDialog />);
    expect(component).toMatchSnapshot();
  });
});
