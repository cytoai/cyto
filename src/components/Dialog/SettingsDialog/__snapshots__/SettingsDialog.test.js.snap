// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsDialog should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
